docs: Clean up install_linux with pip

lamberta opened the pull request:
Cleans up the 'How to Install TensorFlow' section using pip in virtualenv and system. Uses virtualenv in a typical workflow.
Fixes #18767

lamberta committed:
docs: Clean up install_linux with pip

lamberta commented:
Staged: go/tfo-stage/install/install_linux

MarkDaoust commented on tensorflow/docs_src/install/install_linux.md:
```diff
@@ -103,248 +103,203 @@ the specified versions. If upgrading is not possible, then you may still run
-     by issuing a command of the following format:
+<pre class="prettyprint lang-bsh">
+  <code># Select one:</code>
+  <code class="devsite-terminal">sudo pip install tensorflow      # Python 2.7 CPU (no GPU support)</code>
```
Should we mention "tf-nightly" here (or earlier)?

(should we in general across all the install pages?)

lamberta replied:
Added here and above (reworded this section a bit)

MarkDaoust commented on tensorflow/docs_src/install/install_linux.md:
```diff
@@ -103,248 +103,203 @@ the specified versions. If upgrading is not possible, then you may still run
-    $ <b>sudo apt-get install python3-pip python3-dev python-virtualenv</b> # for Python 3.n</pre>
-
-  2. Create a Virtualenv environment by issuing one of the following commands:
+### Use `pip` in a virtual environment
```
Isn't this one heading too many?

lamberta replied:
Intentionally increased subheading on these to simplify the right nav. (displays h2 and h3, but not the h4s below)

lamberta commented on tensorflow/docs_src/install/install_linux.md:
```diff
@@ -103,248 +103,203 @@ the specified versions. If upgrading is not possible, then you may still run
 
-If you encounter installation problems, see
-[Common Installation Problems](#common_installation_problems).
+<pre class="prettyprint lang-bsh">
```
Fixed. Added easy_install as option in step 1 (instead of a Ubuntu package). No need for easy_install here

lamberta commented on tensorflow/docs_src/install/install_linux.md:
```diff
@@ -103,248 +103,203 @@ the specified versions. If upgrading is not possible, then you may still run
-### Next Steps
+<pre class="prettyprint lang-bsh">
+(venv)$ pip install --upgrade tensorflow      # for Python 2.7</code>
+(venv)$ pip3 install --upgrade tensorflow     # for Python 3.n</code>
```
fixed

lamberta commented on tensorflow/docs_src/install/install_linux.md:
```diff
@@ -103,248 +103,203 @@ the specified versions. If upgrading is not possible, then you may still run
-### Prerequisite: Python and Pip
+See the
+[REQUIRED_PACKAGES section of setup.py](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/pip_package/setup.py)
+for a list of TensorFlow packages that `pip` installs or upgrade`.
```
fixed

lamberta commented on tensorflow/docs_src/install/install_linux.md:
```diff
@@ -103,248 +103,203 @@ the specified versions. If upgrading is not possible, then you may still run
-of pip or pip3.  If Version 8.1 or later is not installed, issue the
-following command, which will either install or upgrade to the latest
-pip version:
+On Ubuntu, Python is automatically installed and `pip` is *usually* installed.
```
added above

lamberta commented on tensorflow/docs_src/install/install_linux.md:
```diff
@@ -103,248 +103,203 @@ the specified versions. If upgrading is not possible, then you may still run
+  <code class="devsite-terminal">pip -V  # or: pip3 -V</code>
 </pre>
 
+We *strongly recommend* `pip` or `pip3` version 8.1 or higher. If using a release
```
fixed

lamberta committed:
Review fixes to install_linux

lamberta reviewed (commented):
Addressed comments in latest commit.
Staged latest

MarkDaoust commented on tensorflow/docs_src/install/install_linux.md:
```diff
@@ -116,23 +116,47 @@ There are a few options to install TensorFlow on your machine:
+Confirm the `python` and `pip` versions:
+
+<pre class="prettyprint lang-bsh">
+  <code class="devsite-terminal">python -V</code>
```
does it need a python3 option here as well?

MarkDaoust reviewed (dismissed):
One minor question.
LGTM.

lamberta committed:
docs: install_linux, move GPU section below install procedures.

lamberta commented:
Updated your suggestion to include python3 -V

Before I leave this doc, I went ahead and moved the GPU section that started the doc below the installation procedures. It seemed a little off putting right at the start, especially since we recommend installing the CPU version to start.
